codewof/templates/programming/question_components/question-card.html

```diff
@@ -17,5 +17,18 @@
         <div class="qc-title">
             {{ question.title }}
         </div>
+        <div class="qc-tags">
+            {% include "programming/question_components/badges/difficulty.html" %}
+            {% for concept in question.concepts.all %}
+                {% if concept.slug == "display-text" or concept.slug == "functions" or concept.slug == "inputs" or concept.slug == "single-condition" or concept.slug == "multiple-conditions" or concept.slug == "advanced-conditionals" or concept.slug == "conditional-loops" or concept.slug == "range-loops" or concept.slug == "string-operations" or concept.slug == "lists" %}
```

Comment:
To only display the most indented filter tag for each question (e.g., the main overarching tag "Conditionals" is not displayed, but instead, "Single Conditional", or an equivalent indented filter is displayed, that is more specific and less general). I could not find a better way to do this, unfortunately (as it's also dependent on the tag, for some there is no sub-tag such as the "Display Text" filter), but if you have any ideas please let me know.

Reply:
Let me first check if I understand the system correctly: For programming concepts there are instances of each concept, so an entry for 'single condition' and one for 'conditionals'. When a question is added to a subcategory, it also gets added to its parent if there is one.

This results in the following two filters showing the exactly same content, might want to think if this is intended behaviour:
![image](https://user-images.githubusercontent.com/8001048/191631639-b0516a6f-2c7d-4e31-a502-7113091c8f13.png)
![image](https://user-images.githubusercontent.com/8001048/191631653-1ae4b08b-2361-4ba2-84f0-eae6c09efc33.png)

If you only want end/leaf nodes on your concept tree being shown, then it might be worth adding a boolean field to the concept that is `True` if it has no children, then filter using that. You could also filter on the reverse of the foreign key, which is `children` and checking it's null. Though the boolean sounds easier to manage.

It may also be good to display the total number of questions returned under a filterset set.

Reply:
Great idea, I've added a boolean to do this as you mentioned.

Regarding the screenshots, that is the intended behaviour, as the alternative caused the sub-categories to be locked/disabled after checking them all (and thus unchecking the parent category causes all of the sub-categories to be unchecked). We could change this behaviour, but I think that would cause inconsistencies in how the parent categories can be checked. What do you think?

Reply:
I think it's fine as it is currently. especially good enough for release. I think in an older version of CS Unplugged, the behaviour was that the parent was automatically selected if all the children were, but there are also downsides to that version too. Happy with how it is currently as it achieves the functionality required.